File: chanlun-backend/src/main/java/com/chanlun/entity/DataSource.java
package com.chanlun.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.Instant;

/**
 * 数据源实体
 * 
 * 存储交易所 API 连接配置信息
 * API Key 和 Secret Key 使用 AES-256 加密存储
 * 
 * @author Chanlun Team
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@TableName("data_source")
public class DataSource {

    /**
     * 主键
     */
    @TableId(type = IdType.AUTO)
    private Long id;

    /**
     * 数据源名称
     */
    private String name;

    /**
     * 交易所类型 (BINANCE, OKX等)
     */
    private String exchangeType;

    /**
     * API Key (AES-256加密存储)
     */
    private String apiKey;

    /**
     * Secret Key (AES-256加密存储)
     */
    private String secretKey;

    /**
     * REST API 基础 URL
     */
    private String baseUrl;

    /**
     * WebSocket URL
     */
    private String wsUrl;

    /**
     * 是否启用代理
     */
    private Boolean proxyEnabled;

    /**
     * 代理类型 (HTTP/SOCKS5)
     */
    private String proxyType;

    /**
     * 代理地址
     */
    private String proxyHost;

    /**
     * 代理端口
     */
    private Integer proxyPort;

    /**
     * 代理用户名
     */
    private String proxyUsername;

    /**
     * 代理密码 (AES-256加密存储)
     */
    private String proxyPassword;

    /**
     * 是否启用
     */
    private Boolean enabled;

    /**
     * 软删除标记
     */
    @TableLogic
    private Boolean deleted;

    /**
     * 删除时间 (UTC)
     */
    private Instant deletedAt;

    /**
     * 创建时间 (UTC)
     */
    @TableField(fill = FieldFill.INSERT)
    private Instant createdAt;

    /**
     * 更新时间 (UTC)
     */
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private Instant updatedAt;
}

File: chanlun-frontend/src/types/datasource.ts
/**
 * 数据源相关类型定义
 */

/**
 * 交易所类型
 */
export type ExchangeType = 'BINANCE' | 'OKX' | 'HUOBI'

/**
 * 代理类型
 */
export type ProxyType = 'HTTP' | 'SOCKS5'

/**
 * 数据源 DTO
 */
export interface DataSource {
  id: number
  name: string
  exchangeType: ExchangeType
  baseUrl: string
  wsUrl: string
  proxyEnabled: boolean
  proxyType?: ProxyType
  proxyHost?: string
  proxyPort?: number
  proxyUsername?: string
  enabled: boolean
  createdAt: string
  updatedAt: string
  hasApiKey: boolean
  hasSecretKey: boolean
  hasProxyPassword: boolean
}

/**
 * 创建数据源请求
 */
export interface DataSourceCreateRequest {
  name: string
  exchangeType: ExchangeType
  apiKey?: string
  secretKey?: string
  baseUrl?: string
  wsUrl?: string
  proxyEnabled?: boolean
  proxyType?: ProxyType
  proxyHost?: string
  proxyPort?: number
  proxyUsername?: string
  proxyPassword?: string
}

/**
 * 更新数据源请求
 */
export interface DataSourceUpdateRequest {
  name?: string
  exchangeType?: ExchangeType
  apiKey?: string
  secretKey?: string
  baseUrl?: string
  wsUrl?: string
  proxyEnabled?: boolean
  proxyType?: ProxyType
  proxyHost?: string
  proxyPort?: number
  proxyUsername?: string
  proxyPassword?: string
  clearApiKey?: boolean
  clearSecretKey?: boolean
  clearProxyPassword?: boolean
}

/**
 * 数据源分页结果
 */
export interface DataSourcePage {
  records: DataSource[]
  total: number
  size: number
  current: number
  pages: number
}

/**
 * 代理测试结果
 */
export interface ProxyTestResult {
  success: boolean
  message: string
  latencyMs?: number
  statusCode?: number
  responseBody?: string
  testUrl?: string
  errorDetail?: string
}

/**
 * 连接测试结果
 */
export interface ConnectionTestResult {
  success: boolean
  message: string
  latencyMs?: number
  serverTime?: string
  timeDiffMs?: number
}
